File: week6/sentimental-readability/readability.py
def main():
    print("Text: ", end="")
    text = input()  # getting user input
    grade = round(calculateIndex(text))  # calculating grade

    if (grade < 1):  # printing the result
        print("Before Grade 1")
    else:
        print("Grade ", end="")
        if (grade <= 16):
            print(grade)
        else:
            print("16+")


def calculateIndex(text):
    letters = 0
    sentences = 0
    words = 1
    text = text.lower()
    for i in range(len(text)):  # counting each character in given text
        current = text[i]
        if (current >= "a" and current <= "z"):
            letters += 1
        elif (current == " "):
            words += 1
        elif (current == "." or current == "?" or current == "!"):
            sentences += 1
    return ((0.0588 * letters * 100 / words) - (0.296 * sentences * 100 / words) - 15.8)


main()

File: week2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

double calculateCLIndex(string input);

int main(void)
{
    string input = get_string("Text: "); // getting input from user
    int grade = round(calculateCLIndex(input));

    if (grade < 1)
        printf("Before Grade 1\n");
    else
    {
        printf("Grade ");
        if (grade <= 16)
            printf("%i\n", grade);
        else
            printf("16+\n");
    }
}

double calculateCLIndex(string input)
{
    int letters = 0;
    int sentences = 0;
    int words = 1; // necessary variables
    for (int i = 0; i < strlen(input); i++)
    {
        char current = tolower(input[i]);
        if (current <= 'z' && current >= 'a') // if character belongs to letters
            letters++;
        else if (current == ' ') // if character is a space (another word)
            words++;
        else if (current == '.' || current == '?' || current == '!') // if character ends a sentence
            sentences++;
    }
    return 0.0588 * (double) letters * 100 / (double) words -
           0.296 * (double) sentences * 100 / (double) words - 15.8;
}
